File: prql-php/src/SourceLocation.php
<?php

declare(strict_types=1);

namespace Prql\Compiler;

/**
 * Location within a source file.
 */
final class SourceLocation
{
    /**
     * Start line.
     */
    public int $start_line;

    /**
     * Start column.
     */
    public int $start_col;

    /**
     * End line.
     */
    public int $end_line;

    /**
     * End column.
     */
    public int $end_col;
}

File: prqlc/bindings/dotnet/PrqlCompiler/SourceLocation.cs
namespace Prql.Compiler;

/// <summary>
/// Location within a source file.
/// </summary>
/// <param name="StartLine">Start line.</param>
/// <param name="StartCol">Start column.</param>
/// <param name="EndLine">End line.</param>
/// <param name="EndCol">End column.</param>
public readonly record struct SourceLocation(
    ulong StartLine,
    ulong StartCol,
    ulong EndLine,
    ulong EndCol);
